include/bl_storage.h

```diff
@@ -7,15 +7,68 @@
 #ifndef BL_STORAGE_H_
 #define BL_STORAGE_H_
 
-#include <zephyr/types.h>
 #include <string.h>
+#include <zephyr/types.h>
+#include <drivers/nrfx_common.h>
+#include <nrfx_nvmc.h>
+#include <errno.h>
+#include <pm_config.h>
+
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-
 #define EHASHFF 113 /* A hash contains too many 0xFs. */
+#define EREADLCS 114 /* LCS field of OTP is in an invalid state */
+#define EINVALIDLCS 115 /* Invalid LCS*/
+
+/* We truncate the 32 byte sha256 down to 16 bytes before storing it */
+#define SB_PUBLIC_KEY_HASH_LEN 16
+
+/** Storage for the PRoT Security Lifecycle state, that consists of 4 states:
+ *  - Device assembly and test
+ *  - PRoT Provisioning
+ *  - Secured
+ *  - Decommissioned
+ *  These states are transitioned top down during the life time of a device.
+ *  The Device assembly and test state is implicitly defined by checking if
+ *  the provisioning state wasn't entered yet.
+ *  This works as ASSEMBLY implies the OTP to be erased.
+ */
+struct life_cycle_state_data {
+	uint16_t provisioning;
+	uint16_t secure;
+	/* Pad to end the alignment at a 4-byte boundary as the UICR->OTP
+	 * only supports 4-byte reads. We place the reserved padding in
+	 * the middle of the struct in case we ever need to support
+	 * another state.
+	 */
+	uint16_t reserved_for_padding;
+	uint16_t decommissioned;
+};
```

Comment:
Although it's not part of the API, the serialization format supports adding more counters. Change counter_collection::num_counters to 2, and add a monotonic_counter at the end. You can create a new define COUNTER_DESC_LCS to use with monotonic_counter::description. This will use existing infrastructure, and give you much more flexbility wrt. the number of states to support.

I realize it might be a big change, but I wanted to mention it.

Reply:
Will keep this in mind for the next time this is re-written.